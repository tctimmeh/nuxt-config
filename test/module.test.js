const { Nuxt, Builder } = require('nuxt')
const request = require('request-promise-native')

const config = require('./fixture/nuxt.config')

const url = path => `http://localhost:3000${path}`
const get = path => request(url(path))

describe('Server renders page with server config values', () => {
  let nuxt

  beforeAll(async () => {
    nuxt = new Nuxt(config)
    await new Builder(nuxt).build()
    await nuxt.listen(3000)
  }, 60000)

  afterAll(async () => {
    await nuxt.close()
  })

  test('render', async () => {
    const html = await get('/')
    expect(html).toMatchSnapshot()
  })
})
