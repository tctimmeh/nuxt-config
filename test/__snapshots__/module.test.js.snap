// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server renders page with server config values render 1`] = `
"<!doctype html>
<html data-n-head-ssr data-n-head=\\"\\">
  <head data-n-head=\\"\\">
    <style data-vue-ssr-id=\\"17cfdfa9:0\\">.nuxt-progress{position:fixed;top:0;left:0;right:0;height:2px;width:0;opacity:1;transition:width .1s,opacity .4s;background-color:#000;z-index:999999}.nuxt-progress.nuxt-progress-notransition{transition:none}.nuxt-progress-failed{background-color:red}</style><style data-vue-ssr-id=\\"f74d85d8:0\\">pre{display:inline}</style>
  </head>
  <body data-n-head=\\"\\">
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h1>Nuxt-Config Test Page</h1> <p>For each of the values below:</p> <ul><li>Ensure the value shown here is correct</li> <li>Ensure the server-rendered value in the HTML here is correct</li></ul> <h2>Shared Config Value</h2> <p>
    This value is set in the global config and then shared with the client using <code>defer()</code>. Both values
    should be the same.
  </p> <p><strong>Expected Client</strong>: simple config value</p> <p><strong>Expected Server</strong>: simple config value</p> <p><strong>Rendered Value</strong>: simple config value</p> <hr> <h2>Overridden on Client</h2> <p>
    This value is different on server and client. For example, this could be useful for specifying a public URL for an
    API server in the browser and a private, internal URL for the server.
  </p> <p><strong>Expected Client</strong>: isomorphic CLIENT</p> <p><strong>Expected Server</strong>: isomorphic SERVER</p> <p><strong>Rendered Value</strong>: isomorphic SERVER</p> <hr> <h2>Shared Environment Override</h2> <p>
    This value is overridden in the run-time environment and shared between client and server. Both values should be
    the same
  </p> <p><strong>Expected Client</strong>: environment OVERRIDE</p> <p><strong>Expected Server</strong>: environment OVERRIDE</p> <p><strong>Rendered Value</strong>: environment OVERRIDE</p> <hr> <h2>Unique Environment Override</h2> <p>
    This value is overridden in the run-time environment using different variables for client and server.
  </p> <p><strong>Expected Client</strong>: env CLIENT</p> <p><strong>Expected Server</strong>: env SERVER</p> <p><strong>Rendered Value</strong>: env SERVER</p> <hr> <h2>Global Access</h2> <p>
    This is the same as &quot;Overridden on Client&quot; except that the value is read using the global config access instead of
    the component $config plugin.
  </p> <p><strong>Expected Client</strong>: isomorphic CLIENT</p> <p><strong>Expected Server</strong>: isomorphic SERVER</p> <p><strong>Rendered Value</strong>: isomorphic SERVER</p> <hr> <p>
    The END.
  </p></div></div></div><script>window.__NUXT__={layout:\\"default\\",data:[{}],error:null,serverRendered:true,__CONFIG__:{simpleValue:\\"simple config value\\",isoValue:\\"isomorphic CLIENT\\",sharedEnvValue:\\"environment OVERRIDE\\",envValue:\\"env CLIENT\\"}};</script><script src=\\"/_nuxt/62dc542123540a0812a9.js\\" defer></script><script src=\\"/_nuxt/1b5defd97243cf638523.js\\" defer></script><script src=\\"/_nuxt/53d46bc9c81a91782ba3.js\\" defer></script><script src=\\"/_nuxt/331528a23ef970c3e167.js\\" defer></script>
  </body>
</html>
"
`;
